poc/tests/test_field.py

```diff
@@ -49,6 +49,46 @@ def run_ntt_field_test(self, cls: type[NttField]) -> None:
         # Test generator.
         self.assertTrue(cls.gen()**cls.GEN_ORDER == cls(1))
 
+        for log_n in range(8):
+            self.run_nth_roots_test(cls, log_n)
+            self.run_ntt_test(cls, log_n)
+            self.run_ntt_star_test(cls, log_n)
+
+    def run_nth_roots_test(self, cls: type[NttField], log_n: int) -> None:
+        ONE = cls(1)
+        # Tests the first root of unity is one.
+        self.assertEqual(cls.nth_root(0), ONE)
+        # Tests that the n-th root of unity has the right order.
+        self.assertEqual(cls.nth_root(log_n)**(1 << log_n), ONE)
+        # Tests that the n-th root of unity is not of a lower order.
+        for log_k in range(log_n):
+            self.assertNotEqual(cls.nth_root(log_n)**(1 << log_k), ONE)
+
+    def run_ntt_test(self, cls: type[NttField], log_n: int) -> None:
+        # Tests that NTT(P) is the same as evaluating the polynomial P
+        # on the powers of an n-th root of unity.
+        n = 1 << log_n
+        p_mon = cls.rand_vec(n)  # a random polynomial in the monomial basis.
+        got = cls.ntt(p_mon, n)
+
+        root = cls.nth_root(log_n)
+        want = [poly_eval(cls, p_mon, root**i) for i in range(n)]
+        self.assertEqual(got, want, f"log_n: {log_n} p_mon: {p_mon}")
+
+    def run_ntt_star_test(self, cls: type[NttField], log_n: int) -> None:
+        # Tests that NTT*(P) is the same as evaluating the polynomial P
+        # on the powers of an n-th root of unity times an 2n-th root of
+        # unity.
```

Comment:
Thanks for the explanation. Let's try to find a more meaningful name for this algorithm than `ntt_star`. How would you explain the purpose of this function?